### ISSUE TITLE
Data storage / ingress

### ISSUE THREAD

**Issue by karpet** [BlueprintKansas/ksvotes.org#3]
<a href="https://github.com/jakelowen"><img src="https://avatars0.githubusercontent.com/u/5368008?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [jakelowen](https://github.com/jakelowen)**
_Wednesday Feb 07, 2018 at 23:12 GMT_
_Originally opened as https://github.com/gpsimpact/open_voter_reg/issues/4_

----

By default dev and test data is stored in local ephemeral data components. No real security necessary here other than keeping sensitive data and passwords out of code commits. 

For live production data we should decide between 2 approaches.
A) Use nanobox protected data components which are protected by default against ingress via the policies outlined here: https://docs.nanobox.io/data-management/managing-live-data/
In short to access nanobox data components from outside the app environment, you must establish an SSL tunnel.  Pro: good security. Con: May be too hard for non-devs to directly query the DB.

B) We might also consider hosting the DB directly with AWS or google cloud and offload management to the pros. We'd then need to follow their best practices around ingress/ security (i.e. VPCs and so on).

For now we can keep working with local dev containers, but we should explore this further before we go into production.

**Comment by karpet** [comment]
Since we are not using nanobox:

* We'll use PostgreSQL on Heroku or AWS for persistent storage.
* We'll use either PostgreSQL or Redis for ephemeral session storage.

**Comment by karpet** [comment]
Good crypto lib to use https://cryptography.io/en/latest/
